packages/spiderfier/services/managers/spider-manager.ts

```diff
@@ -0,0 +1,175 @@
+        }
+      }
+    });
+  }
+
+  setMarkersWontMove(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.markersWontMove = typeof c.markersWontMove !== 'undefined' ? c.markersWontMove : false;
+    });
+  }
+
+  setMarkersWontHide(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.markersWontHide = typeof c.markersWontHide !== 'undefined' ? c.markersWontHide : false;
+    });
+  }
+
+  setBasicFormatEvents(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.basicFormatEvents = typeof c.basicFormatEvents !== 'undefined' ? c.basicFormatEvents : false;
+    });
+  }
+
+  setKeepSpiderfied(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.keepSpiderfied = typeof c.keepSpiderfied !== 'undefined' ? c.keepSpiderfied : false;
+    });
+  }
+
+  setIgnoreMapClick(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.ignoreMapClick = typeof c.ignoreMapClick !== 'undefined' ? c.ignoreMapClick : false;
+    });
+  }
+
+  setNearbyDistance(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.nearbyDistance = typeof c.nearbyDistance === 'number' ? c.nearbyDistance : 20;
+    });
+  }
+
+  setCircleSpiralSwitchover(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.circleSpiralSwitchover = typeof c.circleSpiralSwitchover === 'number' ? c.circleSpiralSwitchover : 9;
+    });
+  }
+
+  setCircleFootSeparation(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.circleFootSeparation = typeof c.circleSpiralSwitchover === 'number' ? c.circleFootSeparation : 23;
+    });
+  }
+
+  setCircleStartAngle(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.circleStartAngle = typeof c.circleSpiralSwitchover === 'number' ? c.circleStartAngle : Math.PI / 6;
+    });
+  }
+
+  setSpiralFootSeparation(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.spiralFootSeparation = typeof c.circleSpiralSwitchover === 'number' ? c.spiralFootSeparation : 26;
+    });
+  }
+
+  setSpiralLengthStart(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.spiralLengthStart = typeof c.circleSpiralSwitchover === 'number' ? c.spiralLengthStart : 11;
+    });
+  }
+
+  setSpiralLengthFactor(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.spiralLengthFactor = typeof c.circleSpiralSwitchover === 'number' ? c.spiralLengthFactor : 4;
+    });
+  }
+
+  setLegWeight(c: AgmMarkerSpider): void {
+    this._spiderInstance.then(spider => {
+      spider.legWeight = typeof c.circleSpiralSwitchover === 'number' ? c.legWeight : 1;
+    });
+  }
+
+  createEventObservable<T>(eventName: string, marker: AgmMarker): Observable<T> {
+    // Override the default "click" event with "spider_click"
```

Comment:
Could it be a valid use case that the user want's a click event instead of a spider_click event?

Reply:
The click event in the spiderfier is not very functional since it will also trigger when the "root" marker is clicked, long story short, the click event might be usefull for only very few people and to (properly) support the info window, I have to replace the click event (otherwise the infowindow will also trigger when clicking the still collapsed markers)...

If you (or anyone) has a suggestion I'm all ears...